uaclient/entitlements/fips.py

```diff
@@ -311,9 +311,13 @@ def application_status(
             return super_status, super_msg
 
         if os.path.exists(self.FIPS_PROC_FILE):
-            self.cfg.remove_notice(
-                "", messages.FIPS_SYSTEM_REBOOT_REQUIRED.msg
-            )
+            # We are now only removing the notice if there is no reboot
+            # required information regarding the fips metapackage we install.
+            if not util.should_reboot(set(self.packages)):
+                self.cfg.remove_notice(
+                    "", messages.FIPS_SYSTEM_REBOOT_REQUIRED.msg
+                )
+
```

Comment:
Could we actually move this up and replace the call above?
```diff
+        # We are now only removing the notice if there is no reboot
+        # required information regarding the fips metapackage we install.
+        if not util.should_reboot(set(self.packages)):
+            self.cfg.remove_notice(
+                "", messages.FIPS_SYSTEM_REBOOT_REQUIRED.msg
+            )

         if util.is_container() and not util.should_reboot():
-            self.cfg.remove_notice(
-                "", messages.FIPS_SYSTEM_REBOOT_REQUIRED.msg
-            )
             return super_status, super_msg

         if os.path.exists(self.FIPS_PROC_FILE):
-            # We are now only removing the notice if there is no reboot
-            # required information regarding the fips metapackage we install.
-            if not util.should_reboot(set(self.packages)):
-                self.cfg.remove_notice(
-                    "", messages.FIPS_SYSTEM_REBOOT_REQUIRED.msg
-                )
```

Reply:
Good call, I have updated the code